substrate/frame/system/src/offchain.rs

```diff
@@ -489,8 +489,14 @@ pub trait CreateSignedTransaction<LocalCall>:
 }
 
 /// Interface for creating an inherent.
+///
+/// It can also be used to create an unsigned transaction as they are both the same extrinsic
+/// variant: `Bare`. Unsigned transaction are deprecated in favor of general transaction.
 pub trait CreateInherent<LocalCall>: CreateTransactionBase<LocalCall> {
```

Comment:
The naming is still just wrong and trying to fix this via docs is not making it better. As I already said in the original pr, the name of the trait is wrong. 

I would propose we introduce a new `CreateBare` with proper docs and tag this trait deprecated.

Reply:
I updated the PR: deprecate `CreateInherent`, introduce `CreateBare` and replace all usage in polkadot-sdk.

It is a breaking change so I am not sure I can backport anything to 2412, but I will backport it to 2503.

I acknowledge my mistake in transaction extension review, overall I underestimated the stability and the pace of new feature in polkadot-sdk in 2024.

As a side note I think offchain traits could be improved as well:
* we could remove `LocalCall` everywhere, people can bound `RuntimeCall: From<Call>` in their config
* `CreateTransactionBase` could simply bound `frame_system::Config` as supertrait instead of having a duplicated associated type `RuntimeCall` that bring name conflict. (We do bound `Config` as supertrait for `SigningTypes`).
But better not to break them.

Reply:
I agree with the offchain trait refactor, but it's probably best to do it after phase 2 of Extrinsic Horizon is done.